frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-overview/experiment-overview.component.ts

```diff
@@ -284,30 +313,54 @@ export class ExperimentOverviewComponent implements OnInit, OnDestroy {
         conditionOrder,
         context,
         designType,
+        assignmentAlgorithm,
+        stratificationFactor,
         tags,
         logging,
       } = this.overviewForm.value;
-      const overviewFormData = {
-        name: experimentName,
-        description: description || '',
-        consistencyRule: consistencyRule,
-        conditionOrder: conditionOrder,
-        assignmentUnit: unitOfAssignment,
-        group: groupType,
-        type: designType,
-        context: [context],
-        tags,
-        logging,
-      };
-      this.emitExperimentDialogEvent.emit({
-        type: eventType,
-        formData: overviewFormData,
-        path: NewExperimentPaths.EXPERIMENT_OVERVIEW,
-      });
+      let stratificationFactorValueToSend = stratificationFactor;
```

Comment:
This function is already overloaded, lines 321 through 336 seem like a good place to slice out the logic into its own function.